react week2 homework finished-Ceylan

ceylanbaysalaydin opened the pull request:
HackYourWeather App Link : https://xenodochial-wing-7532d8.netlify.app/

NoerGitKat commented on week2/react-exercises/src/Joke.jsx:
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+export default function Joke({ joke: { setup, punchline } }) {
```
Nice multiple level destructuring! Try to be this consistent in all your components.

NoerGitKat commented on week2/react-exercises/src/DogPhoto.jsx:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export default function DogPhoto({ url }) {
+  return <img src={url} alt="dog-image" height="250" width="400"></img>;
```
Great simple and reusable component!

NoerGitKat commented on week2/react-exercises/src/DogGallery.jsx:
```diff
@@ -0,0 +1,44 @@
+        }
+      })
+      .then((data) => {
+        setDogPhotos((previousState) => [...previousState, data.message]);
```
Since we're setting the state directly, without need to copy the previous state,  we can simply give only the `data.message` argument to `setDogPhotos`.

NoerGitKat commented on week2/react-exercises/src/DogGallery.jsx:
```diff
@@ -0,0 +1,44 @@
+  const [hasError, setError] = useState(false);
+  const getDogPhoto = () => {
+    setLoading(true);
+    fetch('https://dog.ceo/api/breeds/image/random')
```
Could you rewrite this using `async/await` and `try/catch`?

NoerGitKat commented on week2/react-exercises/src/DogGallery.jsx:
```diff
@@ -0,0 +1,44 @@
+  return (
+    <div>
+      <h1> Exercise 2 - Dog Photo Gallery </h1>
+      {dogPhotos.length === 0 && <h2>Get your first dog by clicking the button!</h2>}
```
You could combine both: if there are no dog photos, display the message (`<h2>Get your first dog by clicking the button!</h2>`). Else, map the dog photos.

NoerGitKat commented on week2/react-exercises/src/RandomJoke.jsx:
```diff
@@ -0,0 +1,39 @@
+  const [buttonCount, setButtonCount] = useState(0);
+  const getRandomJoke = () => {
+    setLoading(true);
+    fetch('https://official-joke-api.appspot.com/random_joke')
```
Could you rewrite with `async/await` and `try/catch`?

NoerGitKat commented on week2/react-exercises/src/Button.jsx:
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+export default function Button({ onClick, title }) {
```
Nice reusable component! There is no need for the `<div>` though.

NoerGitKat commented on week2/project/hackyourweather/src/WeatherCard.jsx:
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+function WeatherCard({ cityWeather }) {
+  if (cityWeather.cod !== 200) {
```
This doesn't cover all error states. Why don't you flip it and check for other status codes (like 400 or 404). Based off that you can give more appropriate user feedback.

NoerGitKat commented on week2/project/hackyourweather/src/Form.jsx:
```diff
@@ -0,0 +1,14 @@
+
+export default function Form({ inputValue, setInputValue, onSubmit }) {
+  return (
+    <div>
```
No need for the `<div>`.

NoerGitKat commented on week2/project/hackyourweather/src/App.js:
```diff
@@ -0,0 +1,55 @@
+  const getCityWeather = () => {
+    if (inputValue !== '') {
+      setLoading(true);
+      fetch(URL)
```
Rewrite with `async/await` and `try/catch`.

wouterkleijn commented:
Thanks a lot for the review @NoerGitKat  ;) @ceylanbaysalaydin you never replied to the comments, could you do this still?

ceylanbaysalaydin commented:
> You're doing great Ceylan! I've put some suggestions in the comments, please have a look.



> Thanks a lot for the review @NoerGitKat ;) @ceylanbaysalaydin you never replied to the comments, could you do this still?

Thanks @NoerGitKat , for very detailed feedback! It is really helpful. I already changed my fetch function to async function in the 3rd week but didn't applied small details like deleting unnecessary div. I will do those also but I think I changed the most important one 😅 Big thanks to you I got lots of errors and I learned a lot about how use effect and async function works together. 

And big thanks to @wouterkleijn ! It is good to see you are following our progress again. I haven't got any comment from you for a while. As always I am late to say thank you here but I prefer replying the feedbacks when I have time to change my code according to feedback at the same time. Anyway I just wanted to inform you that's why I reply late.. But I reply all the feedbacks sooner or later, no worries. Anyway I really appreciate this. Thank you very much again to both of you 🌹
